menu/reader.js

```diff
@@ -0,0 +1,41 @@
+const fs = require("fs");
+const YAML = require("js-yaml");
+
+class Menu {
+    constructor(text) {
+        this.delivery = text['Delivery'];
+        this.instructions = text['Instructions']; // If instructions provided
+        this.dishes = text['Menu'];
+        this.modifiers = text['Modifiers'];
+    }
+}
+
+// Convert .yml file to yml text
+function readYmlFile(filename) {
+    try {
+        const raw = fs.readFileSync(filename, 'utf-8');
+        return raw
+    } catch (e) {
+        console.log(e)
+    }
+}
+
+// Returns menu object
+function getMenu(rawText) {
+    try {
+        const data = YAML.load(rawText)
+        const menu = new Menu(data);
+        return menu;
+    } catch (e) {
+        console.log(e);
+    }
+
+}
+
+// To print the result of the parsed yaml
+console.log(getMenu(readYmlFile('../menu.yml')))
```

Comment:
Let's say there was some error with reading the file - line 19 would have caught it, logged it, then returned a null value (without throwing further error). Then the program will be running getMenu(null). While it does catch the error (line 29), we shouldn't evaluate this expression in the first place. Therefore, line 19 should throw the error upward

Reply:
Ah ok makes sense. Thanks for explaining